documentation/docs/30-advanced/15-websockets.md

```diff
@@ -0,0 +1,135 @@
+
+```js
+import { error } from "@sveltejs/kit";
+
+/** @type {import('@sveltejs/kit').Socket} **/
+export const socket = {
+	upgrade(request) {
+		if (request.headers.get('origin') !== 'allowed_origin') {
+			// Reject the WebSocket connection by throwing an error
+			error(403, 'Forbidden');
+		}
+	}
+};
+```
+
+### Open
+
+The `open` hook is called when a WebSocket connection is opened. It receives the [peer](https://crossws.unjs.io/guide/peer) object, to allow interacting with connected clients, as a parameter.
+
+```js
+/** @type {import('@sveltejs/kit').Socket} **/
+export const socket = {
+	open(peer) {
+		// ...
+	}
+};
+```
+
+### Message
+
+The `message` hook is called when a message is received from the client. It receives the [peer](https://crossws.unjs.io/guide/peer) object, to allow interacting with connected clients, and the [message](https://crossws.unjs.io/guide/message) object, containing data from the client, as parameters.
+
+```js
+/** @type {import('@sveltejs/kit').Socket} **/
+export const socket = {
+	message(peer, message) {
+		// ...
+	}
+};
+```
+
+### Close
+
+The `close` hook is called when a WebSocket connection is closed. It receives the [peer](https://crossws.unjs.io/guide/peer) object, to allow interacting with connected clients, and the close event object, containing the [WebSocket connection close code](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent/code#value) and reason, as parameters.
+
+```js
+/** @type {import('@sveltejs/kit').Socket} **/
+export const socket = {
+	close(peer, event) {
+		// ...
+	}
+};
+```
+
+### Error
+
+The `error` hook is called when a WebSocket connection error occurs. It receives the [peer](https://crossws.unjs.io/guide/peer) object, to allow interacting with connected clients, and the error as parameters.
+
+```js
+/** @type {import('@sveltejs/kit').Socket} **/
+export const socket = {
+	error(peer, error) {
+		// ...
+	}
+};
+```
+
+## Connecting from the client
+
+To connect to a WebSocket endpoint, you can use the [`WebSocket`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket) constructor in the browser.
+
+```svelte
+<script>
+	import { onMount } from 'svelte';
+
+	onMount(() => {
+		// To connect to src/routes/ws/+server.js
+		const socket = new WebSocket(`/ws`);
+
+		socket.addEventListener("open", (event) => {
+			socket.send("Hello Server!");
+		});
+
+		// ...
+	});
+</script>
+```
+
+See [the WebSocket documentation on MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) for more details.
+
+## Compatibility
+
+SvelteKit uses [`unjs/crossws`](https://crossws.unjs.io) to handle cross-platform WebSocket connections. Please refer to their [compatibility table](https://crossws.unjs.io/guide/peer#compatibility) for the `peer` object in different runtime environments.
```

Comment:
The project refers to itself as `crossws` rather than `unjs/crossws` and I think it probably makes sense that we do too

```suggestion
SvelteKit uses [`crossws`](https://crossws.unjs.io) to handle cross-platform WebSocket connections. Please refer to their [compatibility table](https://crossws.unjs.io/guide/peer#compatibility) for the `peer` object in different runtime environments.
```